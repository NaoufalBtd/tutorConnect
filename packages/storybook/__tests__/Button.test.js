import React from 'react';
import renderer from 'react-test-renderer';
// import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import { render } from '@testing-library/react-native';

import MyButtonMeta, {
  Button,
} from '../src/components/Forms/Button/Button.stories';
import Wrapper from '../src/components/Wrapper';

const { argTypes } = MyButtonMeta;

describe('Button component', () => {
  const argTypes = MyButtonMeta.argTypes;

  for (const key in argTypes) {
    const opts = argTypes[key].options;

    if (argTypes[key].type === 'boolean') {
      it(`${key}`, () => {
        //@ts-ignore
        const props = {
          [key]: true,
        };
        const tree = renderer
          .create(
            <Wrapper>
              <Button {...props} />
            </Wrapper>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    } else {
      opts.forEach((arg) => {
        it(`${key} ${arg}`, () => {
          //@ts-ignore
          const props = {
            [key]: arg,
          };
          const tree = renderer
            .create(
              <Wrapper>
                <Button {...props} />
              </Wrapper>
            )
            .toJSON();
          expect(tree).toMatchSnapshot();
        });
      });
    }
  }
});
