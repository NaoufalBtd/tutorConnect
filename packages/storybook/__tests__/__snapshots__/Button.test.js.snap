// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component action negative 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-12s927w 4zwwcu-1qr77c8 4zwwcu-5n5obt 4zwwcu-18jpc09"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-1iooa9 4zwwcu-r3cqrx 4zwwcu-11hczkh 4zwwcu-197pp75"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component action positive 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-10v7flk 4zwwcu-1qr77c8 4zwwcu-5n5obt 4zwwcu-15m3xlk"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-111om69 4zwwcu-r3cqrx 4zwwcu-11hczkh 4zwwcu-1tthglh"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component action primary 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-5n5obt 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-11hczkh 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component action secondary 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-1wlduvr 4zwwcu-1qr77c8 4zwwcu-5n5obt 4zwwcu-14bfgb"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-d2mlpl 4zwwcu-r3cqrx 4zwwcu-11hczkh 4zwwcu-1xgarii"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size lg 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-168ejle 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-16we7vb 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size md 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-5n5obt 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-11hczkh 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size sm 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-k1vcwv 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-1q2dyj6 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size xl 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-5kwyn3 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-1cd3x4v 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size xs 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-14msekw 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-7r3sa2 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component variant link 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1jps7nv 4zwwcu-5n5obt 4zwwcu-14xikch"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-11hczkh"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component variant outline 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1cqu5g 4zwwcu-5n5obt 4zwwcu-4s28py"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-11hczkh"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component variant solid 1`] = `
<div
  className="css-view-175oi2r"
  data-style="1mk50wr-1q64x7k 14bja7p-1q64x7k n7aciq-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="y9czwk-1hh0pc0 14bja7p-1q64x7k yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
      data-style="4zwwcu-6gd0e6 4zwwcu-58zsu5 4zwwcu-1qr77c8 4zwwcu-5n5obt 4zwwcu-upz7x2"
      dir={null}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="css-text-1rynq56"
        data-style="19n9yv0-4du8hr 4zwwcu-1996vl6 4zwwcu-7s6zs0 4zwwcu-r3cqrx 4zwwcu-11hczkh 4zwwcu-1b07oxx"
        dir="auto"
      >
        Button
      </div>
    </div>
  </div>
</div>
`;
